fog/overseer/server/src/worker.rs

```diff
@@ -0,0 +1,544 @@
+    is_enabled: Arc<AtomicBool>,
+
+    /// If this is true, the thread will stop.
+    stop_requested: Arc<AtomicBool>,
+
+    logger: Logger,
+}
+
+/// This associates an IngestSummary with an IngestClient. This makes it easy
+/// to query a given node based on its IngestSummary.
+struct IngestSummaryNodeMapping {
+    node_index: usize,
+
+    ingest_summary: IngestSummary,
+}
+
+impl<DB: RecoveryDb> OverseerWorkerThread<DB>
+where
+    OverseerError: From<DB::Error>,
+{
+    /// Poll the Fog Ingest cluster every 5 seconds.
+    const POLLING_FREQUENCY: Duration = Duration::from_secs(5);
+
+    /// Try a request to Fog Ingest node this many times if you encounter an
+    /// error.
+    const NUMBER_OF_TRIES: u64 = 3;
+
+    pub fn start(
+        ingest_clients: Vec<FogIngestGrpcClient>,
+        recovery_db: DB,
+        is_enabled: Arc<AtomicBool>,
+        stop_requested: Arc<AtomicBool>,
+        logger: Logger,
+    ) {
+        let thread = Self {
+            ingest_clients,
+            recovery_db,
+            is_enabled,
+            stop_requested,
+            logger,
+        };
+        thread.run();
+    }
+
+    fn run(self) {
+        loop {
+            log::trace!(self.logger, "Overseer worker start of thread.");
+            std::thread::sleep(Self::POLLING_FREQUENCY);
+
+            if self.stop_requested.load(Ordering::SeqCst) {
+                log::info!(self.logger, "Overseer worker thread stopping.");
+                break;
+            }
+
+            if !self.is_enabled.load(Ordering::SeqCst) {
+                log::trace!(self.logger, "Overseer worker is currently disabled.");
+                continue;
+            }
+
+            let ingest_summary_node_mappings: Vec<IngestSummaryNodeMapping> = match self
+                .retrieve_ingest_summary_node_mappings()
+            {
+                Ok(ingest_summary_node_mappings) => ingest_summary_node_mappings,
+                Err(err) => {
+                    log::error!(self.logger, "Encountered an error while retrieving ingest summaries: {}. Returning to beginning of overseer logic.", err);
+                    continue;
+                }
+            };
+
+            // TODO: Use these ingest summaries to send the desired metadata
+            // to Prometheus: number of keys, number of active nodes, etc.
+            let active_ingest_summary_node_mappings: Vec<&IngestSummaryNodeMapping> =
+                ingest_summary_node_mappings
+                    .iter()
+                    .filter(|ingest_summary_node_mapping| {
+                        ingest_summary_node_mapping.ingest_summary.mode
+                            == IngestControllerMode::Active
+                    })
+                    .collect();
+
+            let active_node_count = active_ingest_summary_node_mappings.len();
+            match active_node_count {
+                0 => {
+                    log::error!(
```

Comment:
Recommend `warn!` here, at best. Situation is abnormal, but no human intervention is required (yet).

Reply:
SGTM, done.